modules/packages/Sort.chpl

```diff
@@ -47,24 +198,27 @@ private inline proc chpl_sort_cmp(a, b, param reverse=false, param eq=false) {
    :type doublecheck: `bool`
    :arg reverse: Sort in reverse numerical order
    :type reverse: `bool`
+   :arg comparator: Record that redefines the comparison mechanism with one of
+      the methods: ``comparator.key(a)`` or ``comparator.compare(a,b)``
 
  */
-proc BubbleSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false) where Dom.rank == 1 {
+proc BubbleSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false, comparator:?rec=emptyRecord) where Dom.rank == 1 && isRecord(rec) {
```

Comment:
* :+1: to deprecating `reverse` and `doublecheck` - Probably worth getting input from the rest of the team on this.
* I like bubble sort in that it's like the *hello world* of sorting. I feel like it's present more for completeness rather than actual usefulness.